pkg/services/folder/folderimpl/folder.go

```diff
@@ -484,8 +484,10 @@ func (s *Service) Delete(ctx context.Context, cmd *folder.DeleteFolderCommand) e
 				return dashboards.ErrFolderAccessDenied
 			}
 
-			if err := s.deleteChildrenInFolder(ctx, dashFolder.OrgID, dashFolder.UID); err != nil {
-				return err
+			if cmd.ForceDeleteRules {
```

Comment:
If I am not mistaken, after this change the flag ForceDeleteRules controls not just rules deletion but also the dashboards deletion because `deleteChildrenInFolder` iterates through all registered in the service (i.e. at this time it's the dashboard service and alerting dbstore).
Can you please clarify if it is your intention?

Reply:
Thanks for the thorough review!

Yes, this is the case and it was my intention. Currently `deleteChildrenInFolder` pretty much does the same thing as [legacyDelete](https://github.com/grafana/grafana/pull/67259/files#diff-62d73bdc0f7fc91a4f8dab955eba4b288815692725c72b81c1f36b87c5f123c0R493), but in a different way (invoking the `DeleteInFolder` methods of registered services). Even when `ForceDeleteRules` is false, dashboards continue to get deleted because we fall back to `legacyDelete`. The purpose of having this duplicated logic is to prepare the groundwork for when improved [folder deletion permissions](https://github.com/grafana/grafana-enterprise/issues/5144) are in place. At that point, we'll be able to start setting `ForceDeleteRules` to true from the frontend.